frontend/src/components/BeslisboomForm.vue

```diff
@@ -68,21 +128,19 @@ const findConclusion = computed(() => {
 })
 
 function handleNextStep(object: Answer | Redirect) {
-  if (object.nextQuestionId) {
-    questionId.value = String(object.nextQuestionId)
-    questionStore.setQuestionId(questionId.value)
-  }
+  questionStore.setQuestionId(object.nextQuestionId ?? null)
   if (object.nextConclusionId) {
-    conclusionId.value = String(object.nextConclusionId)
-    questionId.value = String(null)
+    questionStore.setConclusionId(String(object.nextConclusionId))
   }
+  categoryStore.updateCurrentCategory(currentCategory.value.topic)
 }
 
 async function givenAnswer(answer: Answer) {
   questionStore.addAnswer(questionId.value)
   if (answer.labels) {
     for (let i in answer.labels) {
       questionStore.addLabel(answer.labels[i], questionId.value)
+      questionStore.addLabelByCategory(answer.labels[i], currentCategory.value.topic)
     }
   }
   handleNextStep(answer)
```

Comment:
I think the handleNextSteps(answer) from line 146 should be moved to the else of
if (answer.redirects) {

} else {
  handleNextStep(answer)
}

You now get a null pointer error when an answer has redirects.

Reply:
I moved this comment to the right place in the code.